xc/common/utils/prjxray_arch_import.py

```diff
@@ -116,6 +116,187 @@ def create_synth_io_tiles(complexblocklist_xml, tiles_xml, pb_name, is_input):
     )
 
 
+def create_hetero_synth_io_tiles(complexblocklist_xml, tiles_xml):
+    """ Creates synthetic IO pad tiles used to connect ROI inputs and outputs to the routing network.
+    """
+    pb_xml_in = ET.SubElement(
+        complexblocklist_xml, 'pb_type', {
+            'name': 'SYN-INPAD',
+        }
+    )
+
+    pb_xml_out = ET.SubElement(
+        complexblocklist_xml, 'pb_type', {
+            'name': 'SYN-OUTPAD',
+        }
+    )
+
+    tile_xml = ET.SubElement(tiles_xml, 'tile', {
+        'name': 'SYN-IOPAD',
+    })
+
+    sub_tile_xml_in = ET.SubElement(
+        tile_xml, 'sub_tile', {
+            'name': 'SYN_IN_SUB_TILE',
+            'capacity': '50'
```

Comment:
This makes doesn't make sense to me?  Shouldn't you be taking in a list of in/out tiles, or tile counts?

Reply:
My thought is that it doesn't matter as long as the capacity is higher than the number of synth tiles at the location.  As long as the constraints have the IOs at the correct Z location it should work fine right? Although it is probably more correct to have the capacity match the number of synth IOs that can be placed there.

Reply:
Kind of, but during routing import you will have tiles with no connections, which is weird.  Best to avoid that.  The whole point of heterogeneous tiles is to precisely describe situations like this.

Reply:
More specifically, somewhere in the flow coordinates should be choosen, and propigated to arch_import, routing_import and the IO placement description.

Reply:
Ok that makes sense.  Would like to get the heterogeneous tiles working in the single IO per tile case first and then I will work on this.